carapace-server/src/main/java/org/carapaceproxy/api/ConfigResource.java

```diff
@@ -47,10 +54,40 @@
 public class ConfigResource {
 
     private static final Logger LOG = Logger.getLogger(ConfigResource.class.getName());
+    private static final String CONFIGURATION_DUMPING_FILENAME = "server.dynamic.config";
 
     @javax.ws.rs.core.Context
     ServletContext context;
 
+    @GET
+    public Response dump() {
+        StreamingOutput configStream = (output) -> {
+            try {
+                HttpProxyServer server = (HttpProxyServer) context.getAttribute("server");
+                ConfigurationStore configStore = server.getDynamicConfigurationStore();
+                String lineSeparator = configStore.getProperty("line.separator", "\n");
+                Set<String> props = new TreeSet();
+                configStore.forEach((k, v) -> {
+                    props.add (k + "=" + v);
+                });
+                StringBuilder builder = new StringBuilder();
+                props.forEach(p -> {
+                    builder.append(p);
+                    builder.append(lineSeparator);
+                });
+                byte[] data = builder.toString().getBytes("UTF-8");
+                output.write(data);
+                output.flush();
+            } catch (Exception e) {
+                throw new WebApplicationException("Unable to dump current configuration.");
+            }
+        };
+        LOG.info("Dumped current configuration from API");
+        return Response.ok(configStream, MediaType.APPLICATION_OCTET_STREAM)
```

Comment:
"text/plain" is consistent with the other API methods in this file.
You should simply add the annotation as suggested above and then let this method return a simple java.lang.String